src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -899,6 +974,52 @@ private LfNetwork create(int numCC, int numSC, Network network, List<Bus> buses,
+     * A boundary bus is connected to one or multiple buses that are all in areas different from each other
+     * If a bus is not in any area, and is not a boundary bus, the area interchange control could not be performed and an exception will be thrown
+     * However, it is possible that the flow through this bus is not considered for the interchange flow of some of the areas it is connected to.
+     * In this case, the slack injection of this bus will be shared between those areas to ensure a correct balancing
+     */
+    // TODO add reports
+    private static void checkBusesWithNoArea(LfNetwork lfNetwork, LfNetworkParameters parameters, LoadingContext loadingContext) {
+        if (parameters.isAreaInterchangeControl()) {
+            List<LfBus> handledBuses = lfNetwork.getAreas().stream().flatMap(area ->
+                    area.getBuses().stream())
+                    .toList();
+            List<LfBus> unhandledBuses = lfNetwork.getBuses().stream().filter(b -> !handledBuses.contains(b)).toList();
+            for (LfBus unhandledBus : unhandledBuses) {
+                Set<LfArea> connectedAreas = new HashSet<>();
+                unhandledBus.getBranches().forEach(branch ->
+                    List.of(branch.getBus1(), branch.getBus2()).forEach(connectedBus -> {
+                        if (connectedBus == null) {
+                            throwUnHandledBus(unhandledBus);
+                        }
+                        if (connectedBus != unhandledBus) {
+                            LfArea area = loadingContext.areaBusMap.get(connectedBus);
+                            if (area == null || !connectedAreas.add(area)) {
+                                throwUnHandledBus(unhandledBus);
+                            }
+                        }
+                    })
+                );
+
+                // If the flow through this bus is considered for the interchange of some of the connected areas,
+                // then it has already been added to the Area's externalBusesSlackParticipationFactors map with a factor 0.0
+                // Here we want to share the slack injection of this bus between the areas that are not considering the flow through this bus for their interchange power flow
+                Set<LfArea> areasSharingSlack = connectedAreas.stream().filter(area -> !area.getExternalBusesSlackParticipationFactors().containsKey(unhandledBus)).collect(Collectors.toSet());
+                if (!areasSharingSlack.isEmpty()) {
+                    areasSharingSlack.forEach(area -> area.addExternalBusSlackParticipationFactor(unhandledBus, 1.0 / areasSharingSlack.size()));
+                    LOGGER.info("Bus {} is not in any Area and is connected to Areas: {} but the flow through this bus is not considered for their interchange flow. For a correct balancing, its slack injection will be shared between those areas", unhandledBus.getId(), connectedAreas.stream().map(LfArea::getId).toList());
+                }
```

Comment:
this, and probably the entire method, should be moved outside LfNetworkLoader to be AIC outerloop code (e.g. in `init` part)

Reply:
refactored in 9a9478c80af6e5e216bc90e464be48e7287c92c6